### PR TITLE
refactor!: remove service token logic

### PR DESCRIPTION
**Pull request by obinns-dd**
# refactor!: remove service token logic

Updating this module to conform to the original technical design for Authorisation:
https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3955228806/Mobile+Platform+Common+module+interfaces+for+authenticated+requests

This has involved extracting the service token request logic as it's heavily dependent on having access to subject and refresh tokens.

I think this is the right approach as the STS approach to tokens feels like business logic that is specific to the GOV.UK One Login application and likely does not apply to other apps that may want to consume this networking module.

Partially addresses: https://govukverify.atlassian.net/browse/DCMAW-10102

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with a short description about the feature or update
- [x] Created a `draft` pull request if it is not yet ready for review

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

## Before merging your pull request:
- [x] Ensure that the code coverage and SonarCloud checks have passed
- [x] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [x] Ran the app to ensure that no regressions have been caused by changes during code review.

### CODE REVIEW

File: Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import Foundation
 @testable import Networking
 
-final class MockAuthenticationProvider: AuthenticationProvider {
-  var bearerToken: String = "testBearerToken"
+final class MockAuthorizationProvider: AuthorizationProvider {
+    private(set) var didFetchToken: String?
```
Comment (DylanRJ) [inline Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift:5]: Is the name for this correct according to what we use it for in tests?

Comment (DylanRJ) [reply Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift:5]: `serviceTokenScope` or just `tokenScope`?

Comment (obinns-dd) [reply Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift:5]: How about `didFetchTokenWithScope`?

File: Tests/NetworkingTests/URLRequest+ExtensionsTests.swift [hunk 2]
```diff
@@ -2,37 +2,9 @@
...
-        XCTAssertEqual(body[0], "grant_type=urn:ietf:params:oauth:grant-type:token-exchange")
-        XCTAssertEqual(body[1], "scope=testScope")
-        XCTAssertEqual(body[2], "subject_token=tesSubjectToken")
-        XCTAssertEqual(body[3], "subject_token_type=urn:ietf:params:oauth:token-type:access_token")
-    }
-    
     func test_authorized() throws {
-        let authorizedRequest = sut.authorized(with: "testBearerToken")
+        let url = try XCTUnwrap(URL(string: "https://www.google.com"))
+        let authorizedRequest = URLRequest(url: url).authorized(with: "testBearerToken")
```
Comment (DylanRJ) [inline Tests/NetworkingTests/URLRequest+ExtensionsTests.swift:7]: Worth just using your `.example` URLRequest and removing a line of code?

Comment (obinns-dd) [reply Tests/NetworkingTests/URLRequest+ExtensionsTests.swift:7]: yep, nice!

File: Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift (hunk 1 above)
Comment (DylanRJ) [reply Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift:5]: the `did` implies a Bool test to me but I'm not going to hold this up for that, to your discretion 🙂

**Review by DylanRJ — APPROVED** [review]

File: Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift (hunk 1 above)
Comment (obinns-dd) [reply Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift:5]: `requestedTokenScope`?

File: Tests/NetworkingTests/URLRequest+ExtensionsTests.swift (hunk 2 above)
Comment (obinns-dd) [reply Tests/NetworkingTests/URLRequest+ExtensionsTests.swift]: Done ✅

File: Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift (hunk 1 above)
Comment (DylanRJ) [reply Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift]: perhaps the latter to keep naming consistent 🙂

Comment (obinns-dd) [reply Tests/NetworkingTests/Mocks/MockAuthenticationProvider.swift]: Done ✅

**Review by mcgarj — APPROVED** [review]

**Pull request merged**